scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -1340,7 +1341,33 @@ def __consolidate_checkpoints(self, checkpoints):
             if os.path.isfile(active_checkpoint_path):
                 os.unlink(active_checkpoint_path)
 
-    def __find_and_read_checkpoints(self, warn_on_stale=False):
+    def _get_all_checkpoints(self):  # type: () -> Dict[six.text_type, Any]
+        """
+        Collects all checkpoint states from all running workers and the checkpoints from saved consolidated file.
+        :return: dict of the checkpoints.
+        """
+        checkpoints = {}  # type: Dict[six.text_type, Any]
+
+        # also get the previously consolidated file.
+        consolidated_checkpoint_path = os.path.join(
+            self.__config.agent_data_path, CONSOLIDATED_CHECKPOINTS_FILE_NAME
+        )
+
+        # also read existing checkpoints from the consolidated checkpoint file.
+        previous_checkpoints = self.__read_checkpoint_state(
+            consolidated_checkpoint_path
+        )
+        if previous_checkpoints:
+            checkpoints.update(previous_checkpoints["checkpoints"])
+
+        # get all checkpoints from all files that are currently processed by all workers.
+        for worker in self.workers.values():
+            for worker_session in worker.sessions:
+                checkpoints.update(worker_session.get_full_checkpoints())
```

Comment:
Is there any chance this could throw?

I guess it's unlikely, probably only if a worker was killed or similat.

Reply:
Yeah, something really bad should happen with the worker or with the `multiprocessing.Manager`(which provides the IPC with the worker methods). Something, that we cannot handle properly and in this case, the only thing that we can do is to indicate a bad health check of the agent.